### PR TITLE
find_xyz_cut can take 3d niimg-like object as mask #1193

### PR DESCRIPTION
**Pull request by SylvainLan**
#1193 
I didn't know if you wanted to keep the possibility to have the mask as a ndaray so to be sure I kept it, but can be removed.

### CODE REVIEW

File: nilearn/plotting/find_cuts.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def find_xyz_cut_coords(img, mask=None, activation_threshold=None):
         -----------
         img : 3D Nifti1Image
             The brain map.
-        mask : 3D ndarray, boolean, optional
+        mask : 3D ndarray, boolean or 3D Niimg-like object, optional
```
Comment (KamalakerDadi) [inline nilearn/plotting/find_cuts.py]: This parameter should take only Nifti images and should be renamed from mask -> mask_img.

```mask_img : 3D Nifti1Image, optional```

Should be added a bit of documentation as,  "An optional brain mask, provided mask_img should not be empty."

File: nilearn/plotting/find_cuts.py [hunk 2]
```diff
@@ -64,13 +64,19 @@ def find_xyz_cut_coords(img, mask=None, activation_threshold=None):
         if mask is None:
             mask = not_mask
         else:
-            mask *= not_mask
+            if isinstance(mask, np.ndarray):
+                mask *= not_mask
+            else:
+                mask = check_niimg_3d(mask)
```
Comment (KamalakerDadi) [inline nilearn/plotting/find_cuts.py]: We should be doing this on the top. First, checking if it is a valid Nifti image in 3D and second using _safe_get_data on mask image to convert to data numpy array. Rest should do the job by itself. Tell me if I miss something here ?

 I think this snippet should be added on top after ```data = _safe_get_data(img)``` at line 55.
```python
mask_img=check_niimg_3d(mask_img)
mask = _safe_get_data(mask_img)
```

**Comment by bthirion** [comment]
LGTM. Need to check the failures though.

**Comment by KamalakerDadi** [comment]
Regarding failure: You may have forgotten to address this test function related to mask to be replaced by mask_img. 

```nilearn.plotting.tests.test_find_cuts.test_find_cuts_empty_mask_no_crash```

**Comment by bthirion** [comment]
LGTM.

File: nilearn/plotting/find_cuts.py [hunk 3]
```diff
@@ -54,6 +54,13 @@ def find_xyz_cut_coords(img, mask=None, activation_threshold=None):
     img = check_niimg_3d(img)
     data = _safe_get_data(img)
 
+    # Retrieve optional mask
+    if mask_img is not None:
+        mask_img = check_niimg_3d(mask_img)
+        mask = _safe_get_data(mask_img)
```
Comment (AlexandreAbraham) [inline nilearn/plotting/find_cuts.py]: Do we really have the guarantee that img and mask_img are in the same space? We should at least check it, and at best raise an error or do a resampling if not.

**Comment by KamalakerDadi** [comment]
There's a change of public API. I (@GaelVaroquaux, borrowing @KamalakerDadi's computer) think that it is fine, because people seldom pass a mask, but we should specify it in the whats_new.rst.

**Comment by GaelVaroquaux** [comment]
I added a whats_new and merged.

**Pull request closed without merging**